sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -412,7 +412,10 @@ def _incoming_open(self, channel, frame):
             self._outgoing_open()
             self._set_state(ConnectionState.OPENED)
         else:
-            pass # TODO what now...?
+            self.close(error=AMQPError(
+                condition=ErrorCondition.IllegalState, 
+                description=f"connection is an illegal state: {self.state}"))
+            _LOGGER.error(f"connection is an illegal state: {self.state}")
```

Comment:
When it comes to logging, it's best not to pre-emptively format the string.
It's a small thing, but when using the f-string syntax, the string will be formatted regardless of whether there's a logger attached.
So by using this syntax, while uglier, the string will only be formatted if the logger is actually going to log the statement - which is more performant:
`_LOGGER.error("connection is in an illegal state: %r", self.state)`